test/collection_test.js

```diff
@@ -900,6 +941,42 @@ describe("Collection", () => {
         }));
       });
 
+      describe("incoming changes hook", () => {
+        it("should be called", () => {
+          let hookCalled = false;
+          articles = testCollection({
+            hooks: {
+              "incoming-changes": [
+                function(payload, collection) {
```

Comment:
Ok, so I suggest we don't pass collection as an argument, because most often the user will have an instance at hand already.